solvebio/cli/data.py

```diff
@@ -159,10 +162,63 @@ def _upload_folder(
                     )
                 )
             else:
-                remote_parent = Object.get_by_full_path(
-                    remote_folder_full_path, assert_type="folder"
-                )
-                Object.upload_file(local_file_path, remote_parent.path, vault.full_path)
+                all_files.append((local_file_path, remote_folder_full_path, vault.full_path))
+
+    # Identify which remote folders already exist
+    upload_root_path, _ = Object.validate_full_path(
+        os.path.join(base_remote_path, local_start)
+    )
+    results = GlobalSearch().filter(path__prefix=upload_root_path, type="folder")
```

Comment:
should this block of code be split out into its own function? maybe a few comments to explain what exactly it is doing.

Reply:
Yup! Can do.